### PR TITLE
Face Area Notebook

### PR DESCRIPTION
**Pull request by rajeeja**
https://uxarray--285.org.readthedocs.build/en/285/examples/003-area-calc.html

### CODE REVIEW

**Comment by anissa111** [comment]
@philipc2 I'm taking a look now!

**Comment by anissa111** [comment]
(Sorry wrong button!!)

**Comment by anissa111** [comment]
Yeah, okay so it's nothing to do with this PR specifically. In `contributing.rst`, we use internal links to link to different section headings. One of the links doesn't have a corresponding heading because, I think, we skipped a number. 

image

No idea why it's failing on this now instead of earlier, but there you go!

**Comment by rajeeja** [comment]
Thanks @anissa111 
Had to open the notebook in an editor and remove tensorflow (my local conda env name) in name/display_name `metadata` section of python notebook. 

Seems to be fine now, please review

File: docs/examples/003-area-calc.ipynb [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+    "## Overview\n",
+    "\n",
+    "This notebook will showcase the different area calculation options provided by UXarray\n",
+    "the grid topology attributes and variables stored in the `UXarray.Grid`\n",
+    "object.\n",
+    "\n",
+    "For more details on how to load in data, check out our [previous usage\n",
+    "example](https://uxarray.readthedocs.io/en/latest/examples/read-grid-data.html)\n",
+    "\n",
+    "**This notebook has teh following sections:**\n",
```
Comment (philipc2) [inline docs/examples/003-area-calc.ipynb:21]:
```suggestion
    "**This notebook has the following sections:**\n",
```

File: docs/examples/003-area-calc.ipynb [hunk 2]
```diff
@@ -0,0 +1,360 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "# Calculating face area(s) with UXarray\n",
```
Comment (philipc2) [inline docs/examples/003-area-calc.ipynb:10]: I suggest using a simpler name, maybe the following?

```suggestion
    "# Face Area Calculations\n",
```

File: docs/examples/003-area-calc.ipynb [hunk 3]
```diff
@@ -0,0 +1,360 @@
...
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "## 2. Options for calculate_total_face_area functions\n",
```
Comment (philipc2) [inline docs/examples/003-area-calc.ipynb:117]:
```suggestion
    "## 2. Options for `Grid.calculate_total_face_area` functions\n",
```

File: docs/examples/003-area-calc.ipynb [hunk 4]
```diff
@@ -0,0 +1,360 @@
...
+    "4. Calculate area of a single triangle\n",
+    "5. Calculate area of multiple vertices in spherical coordinates\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "## 1. Load mesh files and calculate total area of all faces\n",
+    "\n",
+    "We will be using outCSne30.ug grid file, that follow the UGRID convention.\n",
```
Comment (philipc2) [inline docs/examples/003-area-calc.ipynb:34]:
```suggestion
    "We will be using the `outCSne30.ug` grid file which is encoded in the UGRID convention.\n",
```

**Review by philipc2 — CHANGES_REQUESTED** [review]
Looks good so far, please give it another read through for typos.

Can you also move major comment blocks into markdown cells for better readability (see our other example notebooks as a reference).

File: docs/gallery.yml [hunk 5]
```diff
@@ -5,3 +5,7 @@
 - title: Accessing Grid Information
   path: examples/002-access-grid-info.ipynb
   thumbnail: _static/thumbnails/default.svg
+
+- title: Calculating face area(s) with UXarray
```
Comment (erogluorhan) [inline docs/gallery.yml:9]: I'd recommend to remove "with UXarray".

File: docs/examples/003-area-calc.ipynb [hunk 6]
```diff
@@ -0,0 +1,360 @@
...
+    "# Face Area Calculations\n",
+    "\n",
+    "## Overview\n",
+    "\n",
+    "This notebook will showcase the different area calculation options provided by UXarray\n",
+    "the grid topology attributes and variables stored in the `UXarray.Grid`\n",
+    "object.\n",
+    "\n",
+    "For more details on how to load in data, check out our [previous usage\n",
+    "example](https://uxarray.readthedocs.io/en/latest/examples/read-grid-data.html)\n",
```
Comment (erogluorhan) [inline docs/examples/003-area-calc.ipynb:19]: This link is broken

File: docs/examples/003-area-calc.ipynb [hunk 7]
```diff
@@ -0,0 +1,360 @@
...
+   "metadata": {
+    "collapsed": false
+   },
+   "source": [
+    "# Face Area Calculations\n",
+    "\n",
+    "## Overview\n",
+    "\n",
+    "This notebook will showcase the different area calculation options provided by UXarray\n",
+    "the grid topology attributes and variables stored in the `UXarray.Grid`\n",
```
Comment (erogluorhan) [inline docs/examples/003-area-calc.ipynb:15]: "... the grid topology ..." may need to be rephrased, was the intent to say something like ""... through grid topology ...""?

File: docs/examples/003-area-calc.ipynb [hunk 8]
```diff
@@ -0,0 +1,360 @@
...
+    "object.\n",
+    "\n",
+    "For more details on how to load in data, check out our [previous usage\n",
+    "example](https://uxarray.readthedocs.io/en/latest/examples/read-grid-data.html)\n",
+    "\n",
+    "**This notebook has the following sections:**\n",
+    "1. Compute total area of all faces\n",
+    "2. Options for accuracy of area calculation functions\n",
+    "3. Getting area of all the faces (not total)\n",
```
Comment (erogluorhan) [inline docs/examples/003-area-calc.ipynb:24]: Maybe say "3. Getting individual areas of each face \n" or "3. Getting areas of all the indivdual faces \n" instead?

File: docs/examples/003-area-calc.ipynb [hunk 9]
```diff
@@ -0,0 +1,360 @@
...
+     "start_time": "2023-04-18T15:29:38.777336Z",
+     "end_time": "2023-04-18T15:29:40.289829Z"
+    }
+   },
+   "outputs": [],
+   "source": [
+    "# Calculate total area of the first (ugrid) mesh\n",
+    "area = ugrid.calculate_total_face_area()\n",
+    "area\n",
+    "# 4 Pi is the area 4*3.14 = 12.56"
```
Comment (erogluorhan) [inline docs/examples/003-area-calc.ipynb:107]: "4*pi"

**Review by erogluorhan — COMMENTED** [review]
Thanks for putting this together! Please see my inline comments.

File: docs/examples/003-area-calc.ipynb [hunk 10]
```diff
@@ -0,0 +1,360 @@
...
+     "end_time": "2023-04-18T15:29:40.344903Z"
+    }
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "verts_grid.face_areas"
```
[the remark this answers is not in the document]
Comment (rajeeja) [reply docs/examples/003-area-calc.ipynb:410]: not a typing errror, this is something that works with 

`
config.DISABLE_JIT = True
`

Numba doesn't like something. Also, there is another issue with providing a grid without z-cords, that also fails.
Debugging this is hard, trying to see what's wrong.

I am planning to introduce a couple more options with helpers that would not rely on UXarray loaded grid to this file:

ux.helpers.calculate_face_area
and
ux. helpers.get_all_face_area_from_coords

**Review by erogluorhan — APPROVED** [review]

**Review by philipc2 — APPROVED** [review]

**Pull request merged**